modules/ZWave/index.js

```diff
@@ -1749,53 +1749,48 @@ ZWave.prototype.parseAddCommandClass = function (nodeId, instanceId, commandClas
 				}
 			}
 
-			if (withTemp && !self.controller.devices.get(deviceNamePrefix + this.CC["ThermostatSetPoint"])) {
+			if (withTemp) {
 				var withTempHeat = instance.ThermostatSetPoint.data[MODE_HEAT],
```

Comment:
The previous code worked perfectly for heat/cool if both are preset (setpoint is used for both modes, switch changes between cool and heat).

For off/heat and off/cool setpoint is used for temperature and switch to turn off/on.

In your case there would be two widgets for setpoint (while only one can be currently active) and how to switch between modes?